gitops/snapshot_integration_tests_status.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions andF
+limitations under the License.
+*/
+
+package gitops
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/api/v1beta1"
+	"github.com/redhat-appstudio/integration-service/helpers"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IntegrationTestStatusDetail contains metadata about the particular scenario testing status
+type IntegrationTestStatusDetail struct {
+	// ScenarioName name
+	ScenarioName string `json:"scenario"`
+	// The status summary for the ITS and Snapshot
+	Status IntegrationTestStatus `json:"status"`
+	// The time of reporting the status
+	Timestamp time.Time `json:"timestamp"`
```

Comment:
Could we have `StartedAt` and `CompletedAt` instead of one `TimeStamp` since there is in checkrun in https://github.com/google/go-github/blob/master/github/checks.go#L30-L31 and i also need it in https://github.com/redhat-appstudio/integration-service/pull/296/files#diff-5d594af46073dcd634d9af0fc3bd8a42698a996c1cb8c2e04d744d18543076e7R514-R516, then users can know the start time and completion time and I can also use it to check if update is needed.

Reply:
I'd keep this as `LastUpdatedAt`, and add `CompletedAt` and `StartedAt` in new PR

Reply:
Please update deign doc, and write where exactly `StartedAt` should be added

Reply:
I added the requirement in a comment [STONEINTG-523](https://issues.redhat.com/browse/STONEINTG-523?focusedId=22936260&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-22936260). 

The requested fields are "StartTime" and "CompletionTime"(Sorry for the name change).

 "StartTime" can be set when we start to handle ITS and update the status to "InProgress", and "CompletionTime" should be set when we finished handling ITS and end with TestFail/TestPassed/DeploymentError/EnvironmentProvisionError.

Thanks.